microcosm_flask/audit.py

```diff
@@ -0,0 +1,98 @@
+"""
+Audit log support for Flask routes.
+
+"""
+from functools import wraps
+from json import loads
+from traceback import format_exc
+
+from flask import current_app, request
+
+from microcosm_flask.errors import extract_status_code, extract_error_message
+
+
+def audit(func):
+    """
+    Record a Flask route function in the audit log.
+
+    Generates a JSON record in the Flask log for every request.
+
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        response = None
+
+        # always include these fields
+        audit_dict = dict(
+            operation=request.url_rule.rule,
+            func=func.__name__,
+            method=request.method,
+        )
+
+        # include request body on debug (if any)
+        if current_app.debug and request.get_json(force=True, silent=True):
```

Comment:
Request and response content are now only sent with debug enabled. It's too big of a privacy risk to log content otherwise.